### PR TITLE
Adding a Panic button

### PR DESCRIPTION
**Pull request by chavonadelal**
## About the PR

Adding a Panic button for the command of the station. The button sends an o signal asking for help indicating the approximate location of the sender.
## Why / Balance

- I think it would be logical for such important people to wear such a button, since they can report a crime and they are not jokers, they listen to their words. The button has a cooldown of 5 seconds between uses to prevent spamming. 

- The antagonists have every chance to confront her using the radio jammer or emp. 

- The button is constantly discharged, so the carriers will need to make sure that it is in working condition (recharge it sometimes) so that it can be used at the right moment. **(This is not working RIGHT NOW and I do not know how to fix it, please help me!)**

- The button is also sensitive to the status of the communication servers, if they do not work, the message will not be sent.

- The panic button is quite unique so it's a great target for a thief!

## Technical details

- A new PanicButtonComponent is used with the system, it is designed for things with a battery. It takes into account whether there is a battery, whether the charge is empty, and popups are sent if there is no battery or it is empty. 

- The button is constantly being discharged. **(This is not working at the moment, please help me, I can't do it without you!)** 😭 

- **Also the popup "comp-panic-button-cell-dead-message" is not sent if the battery is empty, please help me, I do not know how to fix it!** 😢 

## Media

https://github.com/user-attachments/assets/be0b3512-3770-464c-9bdb-e6602ef7ab44

## Requirements

- [x] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: to4no_fix
- add: A new panic button has been added that runs on batteries and, when used, sends a message to the security radio channel asking for help, indicating the approximate location
- add: A panic button has been added to the lockers of the entire station command
- add: The panic button has been added as a new target for theft by a thief

### CODE REVIEW

File: Resources/Prototypes/Catalog/Fills/Lockers/heads.yml [hunk 1]
```diff
@@ -152,6 +157,7 @@
       - id: AccessConfigurator
       - id: RCD
       - id: RCDAmmo
+      - id: HandheldPanicButton
```
Comment (maximus-not) [inline Resources/Prototypes/Catalog/Fills/Lockers/heads.yml:160]: are some of these added 2x once to the hardsuit locker and again to the CE locker again? 
      essentially duplicates?

**Comment by ike709** [comment]
Lol, state-sanctioned `";LING MAINT"`

File: Content.Server/Command/PanicButton/PanicButtonSystem.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            if (!_powerCell.TryGetBatteryFromSlot(panicButton.Owner, out var batteryUid, out var battery, null) &&
+                !TryComp(panicButton.Owner, out battery))
+            {
+                args.Handled = false;
+                return;
+            }
+
+            if (battery.CurrentCharge == 0)
+            {
+                _popup.PopupEntity(Loc.GetString("comp-panic-button-cell-dead-message"), args.User, panicButton.Owner);
```
Comment (NakataRin) [inline Content.Server/Command/PanicButton/PanicButtonSystem.cs:50]: > Also the popup "comp-panic-button-cell-dead-message" is not sent if the battery is empty, please help me, I do not know how to fix it! 😢

I'm not 100% sure that's the correct way of doing it, but gas analyzer works the same way so whynot
```suggestion
                _popup.PopupEntity(Loc.GetString("comp-panic-button-cell-dead-message"), args.User, args.User);
```

Comment (NakataRin) [reply Content.Server/Command/PanicButton/PanicButtonSystem.cs:50]: I think I'll look into everything else a bit later (not that I can help much though)

**Comment by chavonadelal** [comment]
> imo given how its like an immediate "shit get here quick!!" button, cutting the text before the "near" may make it smoother
> 
> "...requested near Captain!"
> 

it is not reported who pressed the button, reported where it was pressed relative to the nearest station beacon, that is why there is the word "near" it could be near the brig  or something else

**Comment by chavonadelal** [comment]
wow someone remembered my pull request 😁

**Comment by chavonadelal** [comment]
- With the advent of AI this is probably no longer needed, but if someone wants to add this to the space-wizards repository I will be only glad, I do not forbid using it for the space-wizards repository
- Also, you can let me know if this is still useful.

**Pull request closed without merging**